common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivityFactory.java

```diff
@@ -96,6 +98,12 @@ public static Intent getAuthorizationActivityIntent(final Context context,
         intent.putExtra(WEB_VIEW_ZOOM_CONTROLS_ENABLED, webViewZoomControlsEnabled);
         intent.putExtra(WEB_VIEW_ZOOM_ENABLED, webViewZoomEnabled);
         intent.putExtra(CORRELATION_ID, DiagnosticContext.INSTANCE.getRequestContext().get(DiagnosticContext.CORRELATION_ID));
+        intent.putExtra(SerializableSpanContext.SERIALIZABLE_SPAN_CONTEXT, SerializableSpanContext.builder()
```

Comment:
I'd suggest that you use JSON based serialization here...at least that's what the class was built with in mind.

Java serialization / de-serialization can be unsafe

Reply:
I created another class to help convert to and from JSON. 51e4f91049ddfd1979ec06cc9c4292590937d561 What do you think?

Reply:
Is there a reason we need a JsonAdapter class specifically for SpanContext?

Can't we write a generic JsonAdapter class that can adapt convert to/from JSON for any Java class? IIRC, I wrote something similar in Linux called as `MoshiJsonAdapter.java`

Reply:
Created CommonMoshiJsonAdapter here: ae60a8e875510a2fb14275ef96f637bfcf3c6083 
I copied the toJson and fromJson methods from your MoshiJsonAdapter.java class. I do think that IJsonAdapter.java should probably be moved to common now, but moving it seemed to affect many LinuxBroker classes, so that's something I can save for a separate PR.